### PR TITLE
Implement good review design

### PR DESCRIPTION
**Pull request by richtier**

### CODE REVIEW

File: caseworker/assets/javascripts/review-good.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    newItems: false,
+    addItemOnBlur: true,
+    filterSetItems: false,
+    addItemsOnPaste: true,
+    minChars: 1,
+    itemName: controlListEntriesField.name,
+    setItems: selected
+  });
+
+  tokenField._renderItem = function() {
```
Comment (richtier) [inline caseworker/assets/javascripts/review-good.js:85]: working around this https://github.com/KaneCohen/tokenfield/blob/master/lib/tokenfield.js#L1366

File: caseworker/cases/views/goods.py [hunk 2]
```diff
@@ -33,6 +37,160 @@ def init(self, request, **kwargs):
         self.action = post_review_goods
         self.success_url = case_url
 
+from django import forms
+
+class ExportControlCharacteristicsForm(forms.Form):
```
Comment (richtier) [inline caseworker/cases/views/goods.py:42]: will move to forms.py

File: caseworker/cases/views/goods.py [hunk 3]
```diff
@@ -34,6 +37,123 @@ def init(self, request, **kwargs):
...
+        # cached in storage. Then later on POSTs and second page the good pks can be retrieved from storage
+        return self.request.GET.getlist(self.object_name) or self.storage.extra_data.get(self.object_name)
+
+    @property
+    def object_pk(self):
+        return self.object_pks[self.steps.index]
+
+    @cached_property
+    def control_list_entries(self):
+        control_list_entries = get_control_list_entries(self.request, convert_to_options=True)
```
Comment (r4vi) [inline caseworker/cases/views/goods.py:61]: would be nice to have a proper cache that works across requests for things like this

File: unit_tests/caseworker/cases/test_views.py [hunk 4]
```diff
@@ -26,116 +26,149 @@ def test_case_audit_trail_system_user(authorized_client, case_pk, queue_pk):
     assert response.status_code == 200
 
 
-# `goods` for SIEL and it's ilk, goods_type for OIEL and it's ilk
-@pytest.mark.parametrize("good_param", ("goods", "goods_types"))
-@pytest.mark.parametrize(
-    "data,expected",
+good_review_parametrize_data = (
```
Comment (r4vi) [inline unit_tests/caseworker/cases/test_views.py:29]: nice work on the parameterized tests

File: caseworker/cases/views/goods.py (hunk 3 above)
Comment (richtier) [reply caseworker/cases/views/goods.py:61]: for this particular view we could cache in `self.strorage.extra_data['clc']`. advantage there is it also cleans itself up once the form wizard has got to the end

Comment (r4vi) [reply caseworker/cases/views/goods.py:61]: I think it's fine for now - no need to introduce complexity for performance problems we don't have yet; just kind of putting down here so we have more generic cache solution in the back of our minds

File: caseworker/templates/case/review-good-base.html [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+                <h2 class="govuk-heading-l">Product {{ wizard.steps.step1 }} of {{ wizard.steps.count }}</h2>
+            {% endif %}
+            {% block good_details_table %}
+            {% endblock %}
+        </div>
+    </div>
+{% endblock %}
+
+{% block javascript %}
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/tarekraafat-autocomplete.js/7.2.0/css/autoComplete.css" integrity="sha512-eP0uk3AYTRMF+HDgLNFW8cFh0mO73gVsnJdgqXH94Zc//fEgtNzMmdyK+cX8qF/SCs7gKUUkeJhJsiTRJstEKA==" crossorigin="anonymous" /><script type="module" src="https://cdnjs.cloudflare.com/ajax/libs/fetch/3.4.0/fetch.min.js" integrity="sha512-+iXlQLHKQZYoFQZfyWkUuJB6X7aZA2+FvAB5PyiYzxRKbgmSLp6vzwXjTlqdvKV5OJS09HHN4lIekb5OOCKhQw==" crossorigin="anonymous"></script>
```
[the remark this answers is not in the document]
Comment (richtier) [reply caseworker/templates/case/review-good-base.html:52]: https://uktrade.atlassian.net/browse/LTD-261

File: caseworker/templates/case/review-good-standard.html [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+                <td class="govuk-table__cell">
+                    {{ object.good.is_military_use.value }}
+                    {% if object.good.modified_military_use_details %}
+                    , {{ object.good.modified_military_use_details }}
+                    {% endif %}
+                </td>
+            </tr>
+
+            <tr class="govuk-table__row app-table__row">
+                <th scope="row" class="govuk-table__header">Component</th>
```
[the remark this answers is not in the document]
Comment (richtier) [reply caseworker/templates/case/review-good-standard.html:58]: probably/unfortunately :laughing: :+1:

File: caseworker/cases/views/goods.py (hunk 3 above)
Comment (richtier) [reply caseworker/cases/views/goods.py]: oops already did it friday :laughing:

**Review by r4vi — APPROVED** [review]

**Pull request merged**